networkx/linalg/tests/test_laplacian.py

```diff
@@ -240,3 +240,142 @@ def test_directed_combinatorial_laplacian():
         nx.directed_combinatorial_laplacian_matrix(G, walk_type="pagerank", alpha=100)
     with pytest.raises(nx.NetworkXError):
         nx.directed_combinatorial_laplacian_matrix(G, walk_type="silly")
+
+
+class TestTotalSpanningTreeWeight:
+    @classmethod
+    def setup_class(cls):
+        global np
+        np = pytest.importorskip("numpy")
+
+    def test_tstw_disconnected(self):
+        G = nx.empty_graph(2)
+        with pytest.raises(nx.NetworkXError):
+            nx.total_spanning_tree_weight(G)
+
+    def test_tstw_no_nodes(self):
+        G = nx.Graph()
+        with pytest.raises(nx.NetworkXPointlessConcept):
+            nx.total_spanning_tree_weight(G)
+
+    def test_tstw_weight(self):
+        # weights are ignored
+        G = nx.Graph()
+        G.add_edge(1, 2, weight=1)
+        G.add_edge(1, 3, weight=1)
+        G.add_edge(2, 3, weight=2)
+        assert np.isclose(nx.total_spanning_tree_weight(G, "weight"), 5)
+
+    def test_tstw_negative_weight(self):
+        # weights are ignored
+        G = nx.Graph()
+        G.add_edge(1, 2, weight=1)
```

Comment:
Why are negative weights ignored instead of an exception raised. Doesn't that cause surprises? It might be better to raise an exception so they know there is a problem and can fix it. Otherwise they may never know they had a negative weight.